Separate TcpKeepAliveIdle and TcpKeepAliveInterval check logic

ValdikSS opened the pull request:
This patch allows to separately configure both parameters.

ValdikSS committed:
Separate TcpKeepAliveIdle and TcpKeepAliveInterval check logic

database64128 commented on transport/internet/sockopt_darwin.go:
```diff
@@ -9,6 +9,8 @@ const (
 	// TCP_FASTOPEN_CLIENT is the value to enable TCP fast open on darwin for client connections.
 	TCP_FASTOPEN_CLIENT = 0x02 // nolint: revive,stylecheck
+	// syscall.TCP_KEEPINTVL is missing on some darwin architectures.
+	sysTCP_KEEPINTVL = 0x101 // nolint: revive,stylecheck
```
Which darwin architectures are missing `TCP_KEEPINTVL`? We are using `x/sys/unix` here, not `syscall`, please double check.

ValdikSS replied:
I've copied this from https://github.com/golang/go/blob/master/src/net/tcpsockopt_darwin.go

database64128 replied:
That's not where you should be looking at.

https://github.com/golang/sys/blob/1d35b9e2eb4edf581781c7f3e2a36fac701f0a24/unix/zerrors_darwin_amd64.go#L1459

https://github.com/golang/sys/blob/1d35b9e2eb4edf581781c7f3e2a36fac701f0a24/unix/zerrors_darwin_arm64.go#L1459

`TCP_KEEPINTVL` is present for both `darwin-amd64` and `darwin-arm64`. You can safely use `unix.TCP_KEEPINTVL`.

xiaokangwang replied:
Your recommendation have been adapted at 406e0f7ac6614e53870687725dfc8e48a142be95.

database64128 commented on transport/internet/sockopt_darwin.go:
```diff
@@ -24,9 +26,16 @@ func applyOutboundSocketOptions(network string, address string, fd uintptr, conf
+			if config.TcpKeepAliveInterval > 0 {
+				if err := unix.SetsockoptInt(int(fd), unix.IPPROTO_TCP, sysTCP_KEEPINTVL, int(config.TcpKeepAliveIdle)); err != nil {
+					return newError("failed to set TCP_KEEPIDLE", err)
+				}
```
I think moving `setsockopt(TCP_KEEPALIVE)` and `setsockopt(TCP_KEEPINTVL)` out of the `if config.TcpKeepAliveIdle > 0 || config.TcpKeepAliveInterval > 0` block would make it more readable.

xiaokangwang replied:
Your recommendation have been adapted at 406e0f7ac6614e53870687725dfc8e48a142be95.

xiaokangwang committed:
adapt style recommendations for keep alive control

From: https://github.com/v2fly/v2ray-core/pull/1484#discussion_r777456773 https://github.com/v2fly/v2ray-core/pull/1484#discussion_r777270291

ValdikSS commented:
@xiaokangwang, thanks for the fix. I'll adapt it for #1483 and update shortly.
